internal/transport/http2_client.go

```diff
@@ -505,9 +507,18 @@ func (t *http2Client) createHeaderFields(ctx context.Context, callHdr *CallHdr)
 		headerFields = append(headerFields, hpack.HeaderField{Name: "grpc-previous-rpc-attempts", Value: strconv.Itoa(callHdr.PreviousAttempts)})
 	}
 
+	advertisedCompressors := t.advertisedCompressors
 	if callHdr.SendCompress != "" {
 		headerFields = append(headerFields, hpack.HeaderField{Name: "grpc-encoding", Value: callHdr.SendCompress})
-		headerFields = append(headerFields, hpack.HeaderField{Name: "grpc-accept-encoding", Value: callHdr.SendCompress})
+		// Advertise the outgoing compressor name, when compressor is not registered
+		// via encoding.RegisterCompressor.
+		if !grpcutil.IsCompressorNameRegistered(callHdr.SendCompress) {
```

Comment:
Is this possible?  How?

Reply:
Clients can use compressors which aren't registered through `encoding.RegisterCompressor` using the deprecated [`WithCompressor`](https://pkg.go.dev/google.golang.org/grpc#WithCompressor) dial option.

Reply:
Does this mean the `AdvertisedCompressors` function should take a single argument instead of a repeated, then?

Or even, remove the parameter altogether, and instead of calling this function again, simply append `","+callHdr.SendCompress` to `advertisedCompressors`?

Reply:
Sounds good, updated